photon-api/src/main/scala/com/linkedin/photon/ml/data/RandomEffectDataSetPartitioner.scala

```diff
@@ -55,53 +56,50 @@ protected[ml] class RandomEffectDataSetPartitioner(idToPartitionMap: Broadcast[M
   }
 
   /**
+   * Hash code for this partitioner.
    *
-   * @return
+   * @return A Int hash code
    */
   override def hashCode: Int = idToPartitionMap.hashCode()
 
   /**
+   * For a given key, get the corresponding partition id. If the key is not in any partition, we randomly assign
+   * the training vector to a partition (with Spark's HashPartitioner).
    *
-   * @return
-   */
-  def numPartitions: Int = partitions
-
-  /**
-   *
-   * @param key
-   * @return
+   * @param key A training vector key (String).
+   * @return The partition id to which the training vector belongs.
    */
   def getPartition(key: Any): Int = key match {
     case string: String =>
-      idToPartitionMap.value.getOrElse(string, defaultPartitioner.getPartition(string))
+      idToPartitionMap.value.getOrElse(string, new HashPartitioner(numPartitions).getPartition(string))
     case any =>
       throw new IllegalArgumentException(s"Expected key of ${this.getClass} is String, but ${any.getClass} found")
   }
-
-  /**
-   *
-   * @return
-   */
-  def defaultPartitioner: HashPartitioner = new HashPartitioner(partitions)
 }
 
 object RandomEffectDataSetPartitioner {
   /**
-   * Generate a partitioner from the random effect dataset.
+   * Generate a partitioner for one type of random effect model.
    *
-   * @param numPartitions Number of partitions
+   * Each RE is for a given "item", so we want to train each RE only on the training vectors that pertain to that
+   * item type. So, for a give RE type, we collect the training vector ids that correspond to that type, then build
+   * an id to partition map. Another concern is to keep the numPartitions as equally loaded as possible, so we do a
+   * simple 'bin packing' (minHeap) to ensure that.
```

Comment:
```
A random effect model is composed of multiple sub-models, each trained on data points for a single item. Data should be distributed across partitions as equally as possible. Since some items have more data points than others, this partitioner uses simple 'bin packing' for distributing data load across partitions.
```
I think that would be easier to understand. Also, we should either use "RE" throughout the whole documentation (at least at single file granularity) or not at all